RGPosadas/LitterRally
# US-2.8: Joining events

This PR closes #66 

To properly test this PR you will need to set up your database such that User with id 1: 
- Is not involved in at least 1 event
 
**Steps to test joining events**
1. Make sure to run `npm i ` on the project to pickup the new dependencies 
2. Run `npm run gen-graphql` to generate the `graphql.tsx`
1. Run `nx serve mull-api` and `nx serve mull-ui`
2. Go to `http://localhost:4200/home/discover`
3. Join an event
3.a By clicking on the join event button on the event card
3.b By clicking on the event card, it brings you to the preview page for the event. Click on "Join" button on the event preview page 

**Steps to run the E2E tests:**
1. Make sure to run `npm i ` on the project to pickup the new dependencies 
1. Run the server `nx serve mull-api`
1. In the root directory run  `nx e2e mull-ui-e2e --watch` ( Cypress takes some time to open up)
